blob/azureblob/azureblob.go

```diff
@@ -816,14 +822,53 @@ func (b *bucket) ListPaged(ctx context.Context, opts *driver.ListOptions) (*driv
 	return page, nil
 }
 
+func (b *bucket) getDelegationCredentials(ctx context.Context) (azblob.StorageAccountCredential, error) {
+	validPeriod := 48 * time.Hour
+	currentTime := time.Now().UTC()
+	expires := currentTime.Add(validPeriod)
+	b.credentialExpiration = expires
```

Comment:
`b.credentialExpiration` should only be updated if the new credentials are successfully updated (i.e., update it at the same time as `b.credentials`.